### PR TITLE
[Coupons] Coupon Details Menu UI

### PR DESCRIPTION
**Pull request by hafizrahman**
Part of: #5533 

### Description

This adds the Compose-based Top App Bar and Dropdown Menu on the Coupon Details screen. It also replaces more references from "coupon summary" to "coupon details".

This does not add functionality for sharing or copying the coupon inside the dropdown menu. It will be added on a separate PR.

### Discussion.

The iOS design of the menu shows a bottom dialog list. In this PR, I opted to use a top-right drop down design instead. This is similar to an existing pattern like on the Product Details screen.

@hichamboushaba Thanks for sharing about ways to set up and modify the nav bar in the fragment. In the end I decided to go ahead and make a Compose-based app bar, for these reasons:
1. It can be a good example pattern if in the future we want to add more and more Compose-based screen,
2. I find the Compose code easier to follow and to work with as everything is in one place, and they're relatively faster to add as well.
3. I also ran into the problem of duplicate top nav bars when trying to modify it in the fragment, and to not spend too much time into it I chose to get back to work on the Compose person.

What do you think?

### Testing instructions

1. Start app, go to Menu, then Coupons.
5. Select a coupon, this will load the Coupon Details screen.
6. Check and ensure the top app bar is shown.
7. Tap the triple-dot menu on top right and make sure two menu items are shown: "Copy Coupon Code" and "Share Coupon".
8. Make sure the dropdown can be opened and closed properly.
9. Tap the back arrow on the app bar and make sure it works as well.

### CODE REVIEW

**Comment by hichamboushaba** [comment]
> @hichamboushaba Thanks for sharing about ways to set up and modify the nav bar in the fragment. In the end I decided to go ahead and make a Compose-based app bar, for these reasons:

Thanks for the ping @hafizrahman, personally, I think a consistent codebase is easier to maintain, and easier for newcomers, that's why I think a decision like this has to be a team-based one, so IMO we should consider the pros and cons and decide what's the best approach, then stick with it for all compose based screens.
If we decide to go with a Compose based toolbar, then it should be good to add it to the guidelines, and add some guidelines on how to use it:
- How to hide the toolbar (I don't have a strong opinion for either approach):
     1. Using `onDestinationChanged` function like what you did (the logic is growing a lot in this function, and things would be harder to maintain soon, and also with the logic you added we are calling `showToolbar` in the `else` bloc, then `hideToolbar` in the `if` you added)
     2. In the Fragment itself by calling `(requireActivity() as MainActivity).hideToolbar()`
- Whether we should create a custom Composable for the toolbar we use to handle different aspects (like theming).

One additional point to be aware of, adding the toolbar to the Fragment instead of the Activity results in a change in animations:
| In Activity | In Fragment |
| ----- | ----- |
| Apr-05-2022 12-12-35| Apr-05-2022 12-19-25 |

Notice how the Toolbar is using the Fragment's transition animation, this might be minor, but we should be aware of it, and its implications later (for example it will be very difficult or even impossible to achieve an animation similar to what was added in https://github.com/woocommerce/woocommerce-android/pull/5371)

@0nko @JorgeMucientes your input would be valuable too 🙏.

> I find the Compose code easier to follow and to work with as everything is in one place.

I agree with this point.

**Comment by 0nko** [comment]
Nice work, @hafizrahman!

> Notice how the Toolbar is using the Fragment's transition animation, this might be minor, but we should be aware of it, and its implications later (for example it will be very difficult or even impossible to achieve an animation similar to what was added in https://github.com/woocommerce/woocommerce-android/pull/5371)

Personally, I like the original transition with the toolbar in place. But I looked at other apps and some do keep the toolbar separate, for example, the Play Store:

https://user-images.githubusercontent.com/1522856/161982964-3fd1b716-425d-4176-996f-9332611e87be.mp4

Maybe if we changed the right-to-left transition to something similar ☝️ then we could keep the Compose toolbar. But I agree with @hichamboushaba, whatever we decide, we should be consistent throughout the app.

**Comment by hichamboushaba** [comment]
To keep conflicts manageable, I'd say let's merge this once the conflicts are fixed, and we can always revisit and adjust according to the team's decision.

**Comment by JorgeMucientes** [comment]
Thanks for handling this @hafizrahman. Looking good! 

I wouldn't block the merging of this PR either, but I'd say let's open a P2 post and discuss the Toolbar addition in Compose screens and the implications it has. Then update the guidelines accordingly (to what comes out of the discussions) and keep a consistent approach (in Inbox for example I've not added a Composed Toolbar). I'd say the 3 points mentioned by @hichamboushaba are a good starting point for the P2 post. 

I'm in favor of adding a Toolbar in Compose and the sooner the better as I believe Compose is going to be our default UI framework in the long-term and sooner or later we are going to have to deal with this.

**Comment by hafizrahman** [comment]
This is good for another review @hichamboushaba 

Thanks for the mention about toolbar change animation as well. The difference still appears here, but I suppose if we decide to refactor back to using the single toolbar method, it will fix the animation issue as well.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt [hunk 1]
```diff
@@ -1,37 +1,71 @@
...
+            backgroundColor = MaterialTheme.colors.surface,
+            title = { Text(state.couponSummary?.code ?: "") },
+            navigationIcon = {
+                IconButton(onClick = { onBackPress() }) {
+                    Icon(Icons.Filled.ArrowBack, contentDescription = "Back")
+                }
+            },
+            actions = {
+                IconButton(onClick = { showMenu = !showMenu }) {
+                    Icon(Icons.Filled.MoreVert, contentDescription = "Coupons Menu")
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt]: Can you set a color to this icon that matches what we have for the other toolbars in the app? currently it's using `action_menu_fg_selector`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt [hunk 2]
```diff
@@ -80,9 +114,10 @@ fun CouponSummarySection(couponSummary: CouponSummaryUi) {
             modifier = Modifier.padding(16.dp)
         ) {
             Text(
-                text = stringResource(id = R.string.coupon_summary_heading),
-                style = MaterialTheme.typography.subtitle1,
+                text = stringResource(id = R.string.coupon_details_heading),
+                style = MaterialTheme.typography.h2,
                 color = MaterialTheme.colors.onSurface,
+                fontSize = 20.sp,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt]: Can you please revert these two changes? I think it's just from the merge conflict.
The correct style is `subtitle1` according to Figma, and with it we don't need to specify font size.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt [hunk 3]
```diff
@@ -117,9 +152,10 @@ private fun CouponPerformanceSection(couponPerformanceState: CouponPerformanceSt
             modifier = Modifier.padding(16.dp)
         ) {
             Text(
-                text = stringResource(id = R.string.coupon_summary_performance_heading),
-                style = MaterialTheme.typography.subtitle1,
+                text = stringResource(id = R.string.coupon_details_performance_heading),
+                style = MaterialTheme.typography.h2,
                 color = MaterialTheme.colors.onSurface,
+                fontSize = 20.sp,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsScreen.kt]: Same as above.

**Review by hichamboushaba — APPROVED** [review]
Works well @hafizrahman, I'm pre-approving, but please make changes for the three comments I left before merging.

**Pull request merged**